tests/TestCase/Error/ErrorTrapTest.php

```diff
@@ -95,6 +96,23 @@ public function testRegisterAndRendering()
         $this->assertStringContainsString('Oh no it was bad', $output);
     }
 
+    public function testRegisterAndHandleFatalUserError()
+    {
+        $trap = new ErrorTrap(['errorRenderer' => TextErrorRenderer::class]);
+        $trap->register();
+        try {
+            ob_start();
+            trigger_error('Oh no it was bad', E_USER_ERROR);
+            $output = ob_get_clean();
+            $this->assertEmpty($output);
```

Comment:
So, this is preventing logging which was happening before?

Reply:
Logging was always happening. What was missing before is that E_USER_ERRORS would be treated in the same way as a warning or notice. The PHP docs say that E_USER_ERROR should be treated like a user-land version of E_ERROR. Historically, these error types are rendered as exceptions and not warnings/notices. Now `ErrorTrap` works the same as the historical error handler did.

Reply:
Guess I don't get what this is testing with the OB output. I see the error is trapped and converted to an exception successfully though.